gamemodes/terrortown/gamemode/server/sv_player_ext.lua

```diff
@@ -299,31 +299,16 @@ end
 -- Adds an @{ITEM} or a @{Weapon} into the bought list of a @{Player}
 -- @note This will disable another purchase of the same equipment
 -- if this equipment is limited
--- @param string cls
+-- @param string equipmentId
 -- @realm server
 -- @see Player:RemoveBought
-function plymeta:AddBought(cls)
+function plymeta:AddBought(equipmentId)
 	self.bought = self.bought or {}
-	self.bought[#self.bought + 1] = tostring(cls)
-
-	BUYTABLE[cls] = true
-
-	net.Start("TTT2ReceiveGBEq")
-	net.WriteString(cls)
-	net.Broadcast()
+	self.bought[#self.bought + 1] = tostring(equipmentId)
 
-	local team = self:GetTeam()
-
-	if team and team ~= TEAM_NONE and not TEAMS[team].alone then
-		TEAMBUYTABLE[team] = TEAMBUYTABLE[team] or {}
-		TEAMBUYTABLE[team][cls] = true
-
-		if SERVER then
-			net.Start("TTT2ReceiveTBEq")
-			net.WriteString(cls)
-			net.Send(GetTeamFilter(team))
-		end
-	end
+	shop.SetEquipmentBought(self, equipmentId)
+	shop.SetEquipmentGlobalBought(equipmentId)
+	shop.SetEquipmentTeamBought(self, equipmentId)
```

Comment:
I can't add a suggestion, but I think here would be `equipmentClass` more fitting as well